src/EventListener/RedirectResponseTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\EventListener;
+
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+trait RedirectResponseTrait
+{
+    public static function getRedirectResponse(RedirectInterface $lastRedirect, ?string $queryString = null): RedirectResponse
```

Comment:
instead of doing this by hand, I'd rather use a library for manipulating URIs. eg. https://uri.thephpleague.com/

there might be use cases that are not handled by this logic. 

eg. merging multiple parameters, or handling urls with anchors.

marketing ppl can be very creative.

Reply:
@loevgaard wdyt about this approach? 

manual implementation or can @next-sentence use URI from The PHP League for URL manipulation instead? my money is on the 2nd option, but we'd need your approval + support to get this merged ASAP

Reply:
I have no problem with introducing a new library. Especially when it's a great library and it's classes/interfaces are not used as type hints in this plugin (because then it will be easily removed later if we have to).